test_files/enum/enum.js

```diff
@@ -27,7 +27,7 @@ enumTestFunction(enumTestValue);
 let /** @type {EnumTest1} */ enumTestLookup = EnumTest1["XYZ"];
 let /** @type {?} */ enumTestLookup2 = EnumTest1["xyz".toUpperCase()];
 // Verify that unions of enum members and other values are handled correctly.
-let /** @type {(boolean|EnumTest1)} */ enumUnionType = EnumTest1.XYZ;
+let /** @type {(EnumTest1|boolean)} */ enumUnionType = EnumTest1.XYZ;
```

Comment:
Is this order stable?  That is, what causes this to have boolean on the right, and will it persist across compiles?

Reply:
Previously, this was ordered by the sequence of types in the resolved union type, handled by TS. Now this is ordered by the lexical spelling, i.e. comes out the way the user wrote this. Both are stable, it's entries in an array.